src/RequireJS/BundleGenerator/ConfigGen.tsx

```diff
@@ -6,17 +6,74 @@
 import * as React from 'react';
 import { RequireConfig, PageModule } from '../../types/require';
 import generate from './generate';
+import JavaScriptHighlighter from '../../JavaScriptHighlighter';
+import styles from './ConfigGen.css';
 
 type Props = {
     pageModules: PageModule[];
     requireConfig: RequireConfig;
 };
 
-export default class ConfigGen extends React.Component<Props> {
-    render() {
+export default class ConfigGen extends React.PureComponent<Props> {
+    generateJS() {
         const { pageModules, requireConfig } = this.props;
         const config = generate(pageModules, requireConfig);
+        const code = JSON.stringify(config, null, 4);
+        return transforms.reduce((code: string, transform) => {
```

Comment:
I don't know if you already know this or not, but if you're immediately returning something, you can just reduce this to:
```tsx
return transforms.reduce((_code, transform) => transform(_code), code);
```
Notice the following too:
1. If `transforms` is properly typed, you don't need to indicate the type of `code: string` in the callback.
2. `code: string` was shadowing the existing `const code` defined on the line just above. I fixed that with a leading underscore. I normally have lint rules against shadowing, FWIW.

Reply:
> If transforms is properly typed, you don't need to indicate the type of code: string in the callback.

Whoops! Was dealing with some type errors and temporarily added that to debug.

> code: string was shadowing the existing const code defined on the line just above. I fixed that with a leading underscore. I normally have lint rules against shadowing, FWIW.

Yep, I knowingly did that. Was the most sensical name for the identifier I could come up with at the time, and didn't need to access `code` in the outer scope.

> I don't know if you already know this or not, but if you're immediately returning something, you can just reduce this to:

Yep! I use implicit return plenty, but in this case the implicit return version blew the line limit for prettier's formatting, and I thought it looked a bit more readable spread out.